File: src/main/java/aurocosh/divinefavor/common/util/UtilSerialize.java
package aurocosh.divinefavor.common.util;

import aurocosh.divinefavor.common.lib.GlobalBlockPos;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

public class UtilSerialize {

    public static int[] serializeGlobalBlockPosArray(List<GlobalBlockPos> posArrayList) {
        int[] array = new int[posArrayList.size() * 4];
        int i = 0;
        for (GlobalBlockPos gpos : posArrayList) {
            array[i++] = gpos.pos.getX();
            array[i++] = gpos.pos.getY();
            array[i++] = gpos.pos.getZ();
            array[i++] = gpos.dimensionId;
        }

        return array;
    }

    public static ArrayList<GlobalBlockPos> deserializeGlobalBlockPosArray(int[] array) {
        int posCount = array.length / 4;
        ArrayList<GlobalBlockPos> posArrayList = new ArrayList<>();

        for (int i = 0, j = 0; i < posCount; i++) {
            int x = array[j++];
            int y = array[j++];
            int z = array[j++];
            int dimension = array[j++];
            posArrayList.add(new GlobalBlockPos(x, y, z, dimension));
        }
        return posArrayList;
    }

    public static UUID stringToUUID(String string) {
        try {
            return UUID.fromString(string);
        }
        catch (IllegalArgumentException var2) {
            return null;
        }
    }
}

File: src/main/java/aurocosh/divinefavor/common/util/UtilSerialize.kt
package aurocosh.divinefavor.common.util

import aurocosh.divinefavor.common.lib.GlobalBlockPos
import java.util.*

object UtilSerialize {

    fun serializeGlobalBlockPosArray(posArrayList: List<GlobalBlockPos>): IntArray {
        val array = IntArray(posArrayList.size * 4)
        var i = 0
        for ((pos, dimensionId) in posArrayList) {
            array[i++] = pos.x
            array[i++] = pos.y
            array[i++] = pos.z
            array[i++] = dimensionId
        }

        return array
    }

    fun deserializeGlobalBlockPosArray(array: IntArray): ArrayList<GlobalBlockPos> {
        val posCount = array.size / 4
        val posArrayList = ArrayList<GlobalBlockPos>()

        var i = 0
        var j = 0
        while (i < posCount) {
            val x = array[j++]
            val y = array[j++]
            val z = array[j++]
            val dimension = array[j++]
            posArrayList.add(GlobalBlockPos(x, y, z, dimension))
            i++
        }
        return posArrayList
    }

    fun stringToUUID(string: String): UUID? {
        try {
            return UUID.fromString(string)
        } catch (var2: IllegalArgumentException) {
            return null
        }

    }
}
